frontend/src/employee-mobile-frontend/components/messages/MessageEditorPage.tsx

```diff
@@ -0,0 +1,137 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React, { useCallback, useContext, useEffect, useState } from 'react'
+import { MessageContext } from '../../state/messages'
+import { UUID } from 'lib-common/types'
+import {
+  MessageReceiversResponse,
+  PostMessageBody
+} from 'lib-common/generated/api-types/messaging'
+import {
+  receiverAsSelectorNode,
+  SelectorNode
+} from 'lib-components/employee/messages/SelectorNode'
+import {
+  deleteDraft,
+  initDraft,
+  postMessage,
+  saveDraft,
+  getReceivers
+} from '../../api/messages'
+import MessageEditor from 'lib-components/employee/messages/MessageEditor'
+import {
+  deleteAttachment,
+  getAttachmentBlob,
+  saveMessageAttachment
+} from '../../api/attachments'
+import { featureFlags } from 'lib-customizations/employee'
+import { useTranslation } from '../../state/i18n'
+import { useParams } from 'react-router-dom'
+import { Result } from 'lib-common/api'
+
+export default function MessageEditorPage() {
+  const { i18n } = useTranslation()
+  const { childId, unitId } = useParams<{
+    unitId: UUID
+    groupId: UUID | 'all'
+    childId: UUID
+  }>()
+  const {
+    nestedAccounts,
+    selectedAccount,
+    selectedUnit,
+    refreshMessages,
+    loadNestedAccounts
+  } = useContext(MessageContext)
+
+  useEffect(() => loadNestedAccounts(unitId), [loadNestedAccounts, unitId])
+
+  const [sending, setSending] = useState(false)
+
+  const [selectedReceivers, setSelectedReceivers] = useState<SelectorNode>()
+
+  useEffect(() => {
+    if (!unitId) return
+
+    void getReceivers(unitId).then(
+      (result: Result<MessageReceiversResponse[]>) => {
+        if (result.isSuccess) {
+          const child = result.value.map(({ receivers }) =>
+            receivers.find((r) => r.childId === childId)
+          )[0]
+          if (!child) return
+          setSelectedReceivers(receiverAsSelectorNode(child))
+        }
+      }
+    )
+  }, [childId, unitId, setSelectedReceivers])
+
+  const onSend = useCallback(
+    (accountId: UUID, messageBody: PostMessageBody) => {
+      setSending(true)
+      void postMessage(accountId, messageBody).then((res) => {
+        if (res.isSuccess) {
+          refreshMessages(accountId)
+          history.back()
+        } else {
+          console.error('Failed to send message')
```

Comment:
PIN login can timeout, so this is not that uncommon. We should have an UI to show this error and maybe instruct to reload / login and try again? At least worth a TODO note